app/uk/gov/hmrc/apiplatform/modules/submissions/domain/models/Submission.scala

```diff
@@ -66,13 +66,81 @@ object Submission {
     def random: Id = Id(UUID.randomUUID().toString())
   }
 
+  val create: (
+      String,
+      Submission.Id,
+      ApplicationId,
+      DateTime,
+      NonEmptyList[GroupOfQuestionnaires],
+      QuestionIdsOfInterest) => Submission = (requestedBy, id, applicationId, timestamp, groups, questionIdsOfInterest) => {
+
+      val initialStatus = Submission.Status.Created(timestamp, requestedBy)
+      val initialInstances = NonEmptyList.of(Submission.Instance(0, Map.empty, NonEmptyList.of(initialStatus)))
+    Submission(id, applicationId, timestamp, groups, questionIdsOfInterest, initialInstances)
+  }
+  
+  val addInstance: (Submission.AnswersToQuestions, Submission.Status) => Submission => Submission = (answers, status) => s => {
+    val newInstance = Submission.Instance(s.latestInstance.index+1, answers, NonEmptyList.of(status))
+    s.copy(instances = newInstance :: s.instances)
+  }
+  
+  val changeLatestInstance: (Submission.Instance => Submission.Instance) => Submission => Submission = delta => s => {
+    s.copy(instances = NonEmptyList(delta(s.instances.head), s.instances.tail))
+  }
+
+  val addStatusHistory: (Submission.Status) => Submission => Submission = newStatus => s => {
+    require(Submission.Status.isLegalTransition(s.status, newStatus))
+    changeLatestInstance(_.copy(statusHistory = newStatus :: s.latestInstance.statusHistory))(s)
+  }
+
+  val changeStatusHistory: (Submission.Status => Submission.Status) => Submission => Submission = delta => s => {
+    val inStatus = s.latestInstance.statusHistory.head
+    val outStatus = delta(inStatus)
+
+    changeLatestInstance(
```

Comment:
does this need a call to isLegalTransition to check the status change makes sense?

Reply:
We can't because answering to answering is not a legal state change.  This is meant to be for manipulating the contents of the current state but you're correct - it could be used badly.  It's only used in one place to change the contents of an answering type.  Probably needs a rewording and a change.